src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/AppointmentService.kt

```diff
@@ -290,16 +291,25 @@ class AppointmentService(
     appointmentDeliveryAddress: AddressDTO? = null,
     npsOfficeCode: String?,
     attended: Attended?,
+    referral: Referral,
     additionalAttendanceInformation: String?,
     notifyProbationPractitioner: Boolean?,
     behaviourDescription: String?,
     appointmentType: AppointmentType,
   ): Appointment {
-    appointment.durationInMinutes = durationInMinutes
-    appointment.appointmentTime = appointmentTime
-    appointment.deliusAppointmentId = deliusAppointmentId
-    setAttendanceAndBehaviourIfHistoricAppointment(appointment, appointmentTime, attended, additionalAttendanceInformation, behaviourDescription, notifyProbationPractitioner, createdByUser, appointmentType)
-    val appointment = appointmentRepository.save(appointment)
+    val appointment = Appointment(
```

Comment:
_Observation_
This change makes `updateAppointment` almost identical to `createAppointment` except for saving the previous appointment with `superseded`.

Should we then instead compose these to avoid duplication?

E.g.
```kt
private fun updateAppointment(...) {
  val newAppointment = createAppointment(...)
  oldAppointment.superseded = true
  appointmentRepository.save(oldAppointment) // do we need saveAndFlush?
  return newAppointment
}
```
or (depending on which order is correct)
```kt
private fun updateAppointment(...) {
  oldAppointment.superseded = true
  appointmentRepository.save(oldAppointment) // do we need saveAndFlush?
  return createAppointment(...)
}
```

Reply:
Refactored this bit.